Apps/Database/src/Delegates/DBGenericCacheDelegate.cs

```diff
@@ -0,0 +1,206 @@
+                                    .Where(p => p.HdId == hdid &&
+                                                p.Domain == domain &&
+                                                p.ExpiryDateTime >= DateTime.UtcNow)
+                                    .OrderByDescending(o => o.CreatedDateTime)
+                                    .FirstOrDefault();
+            if (result.Payload != null)
+            {
+                result.Status = DBStatusCode.Read;
+            }
+
+            return result;
+        }
+
+        /// <inheritdoc />
+        public DBResult<GenericCache> AddCacheObject(GenericCache cacheObject, bool commit = true)
+        {
+            this.logger.LogTrace("Adding GenericCache object to DB...");
+            DBResult<GenericCache> result = new DBResult<GenericCache>()
+            {
+                Payload = cacheObject,
+                Status = DBStatusCode.Deferred,
+            };
+            this.dbContext.GenericCache.Add(cacheObject);
+            if (commit)
+            {
+                try
+                {
+                    this.dbContext.SaveChanges();
+                    result.Status = DBStatusCode.Created;
+                }
+                catch (DbUpdateException e)
+                {
+                    this.logger.LogError($"Unable to save note to DB {e.ToString()}");
+                    result.Status = DBStatusCode.Error;
+                    result.Message = e.Message;
+                }
+            }
+
+            this.logger.LogDebug($"Finished adding Cache object in DB with result {result.Status}");
+            return result;
+        }
+
+        /// <inheritdoc />
+        public DBResult<GenericCache> UpdateCacheObject(GenericCache cacheObject, bool commit = true)
+        {
+            this.logger.LogTrace($"Updating GenericCache request in DB...");
+            DBResult<GenericCache> result = new DBResult<GenericCache>()
+            {
+                Payload = cacheObject,
+                Status = DBStatusCode.Deferred,
+            };
+            this.dbContext.GenericCache.Update(cacheObject);
+            this.dbContext.Entry(cacheObject).Property(p => p.HdId).IsModified = false;
+            if (commit)
+            {
+                try
+                {
+                    this.dbContext.SaveChanges();
+                    result.Status = DBStatusCode.Updated;
+                }
+                catch (DbUpdateConcurrencyException e)
+                {
+                    result.Status = DBStatusCode.Concurrency;
+                    result.Message = e.Message;
+                }
+            }
+
+            this.logger.LogDebug($"Finished updating cache object in DB with result {result.Status}");
+            return result;
+        }
+
+        /// <inheritdoc />
+        public DBResult<GenericCache> DeleteCacheObject(GenericCache cacheObject, bool commit = true)
+        {
+            this.logger.LogTrace($"Deleting GenericCache object from DB...");
```

Comment:
minor nit: the $ isn't needed  here. Myabe it was at one point? @"" would ensure only one copy of this string is held.

Reply:
NIT Resolved.